packages/server/src/context.ts

```diff
@@ -101,29 +101,48 @@ export function closeRequestContext(): void {
   }
 }
 
-function requestIds(req: Request): { requestId: string; traceId: string } {
-  const requestId = randomUUID();
-  const traceIdHeader = req.header('x-trace-id');
-  const traceParentHeader = req.header('traceparent');
-  let traceId: string | undefined;
-  if (traceIdHeader && isUUID(traceIdHeader)) {
-    traceId = traceIdHeader;
-  } else if (traceParentHeader?.startsWith('00-')) {
-    const id = traceParentHeader.split('-')[1];
+const traceIdHeaderMap: {
+  [key: string]: (traceId: string) => string | false;
+} = {
+  'x-trace-id': (traceId) => isUUID(traceId) && traceId,
+  traceparent: (traceId) => {
+    if (!traceId.startsWith('00-')) {
+      return false;
+    }
+
+    const id = traceId.split('-')[1];
     const uuid = [
       id.substring(0, 8),
       id.substring(8, 12),
       id.substring(12, 16),
       id.substring(16, 20),
       id.substring(20, 32),
     ].join('-');
-    if (isUUID(uuid)) {
-      traceId = uuid;
+
+    return isUUID(uuid) && uuid;
+  },
+  // https://github.com/getsentry/sentry-javascript/blob/6dfddc2dfc4b74cbb70dc325b4af165e25904ed4/packages/utils/src/tracing.ts#L117-L130
+  // {uuidv4}-{uuidv4.substring(16)}[-(0|1)]
+  'sentry-trace': (traceId: string) => {
```

Comment:
This is basically a `traceparent` value minus the `version`. The client should just send a valid traceparent value.